### PR TITLE
Add special handling for path cli arg without root path

### LINKED ISSUE #301: Passing a directory from commandline doesn't work with relative paths
Version: `io.elementary.files 0.3.5` (daily repository)

```
$ pwd
/home/martin
$ io.elementary.files Desktop
```
results in trying to open `/Desktop` (see the image), compared to the expected `/home/martin/Desktop`. The same with `./Desktop`, `Desktop/` or `./Desktop/`.

files1

### PR DESCRIPTION
**Pull request by airlanggacahya**
Try to fix #301

### CODE REVIEW

**Comment by jeremypw** [comment]
Thanks for having a go at this.  It seems to work for cli but messes up typing some addresses into the location bar (try "~/Documents").

**Comment by airlanggacahya** [comment]
Rebase to master, and fix the regression.

File: libcore/FileUtils.vala [hunk 1]
```diff
@@ -206,27 +206,58 @@ namespace PF.FileUtils {
...
-                    split_protocol_from_path (sb.str , out scheme, out path);
-                    sb.assign (path);
-                } else if (path.has_prefix ("/../")) {
-                    sb.erase (0, 3);
-                    sb.prepend (get_parent_path_from_path (current_path));
-                    sb.prepend (current_scheme);
-                    split_protocol_from_path (sb.str , out scheme, out path);
-                    sb.assign (path);
+            if (scheme == "" && path.length > 0) {
+                string [] paths = path.split("/", 2);
```
Comment (jeremypw) [inline libcore/FileUtils.vala]: Space before open bracket.

File: libcore/FileUtils.vala [hunk 2]
```diff
@@ -206,27 +206,58 @@ namespace PF.FileUtils {
...
+                    // ignore home documents
+                    case "~":
+                    // ignore path with root
+                    case "":
+                        break;
+                    // process special parent dir
+                    case "..":
+                        sb.assign(current_scheme);
+                        sb.append(Path.DIR_SEPARATOR_S);
+                        sb.append(get_parent_path_from_path (current_path));
```
Comment (jeremypw) [inline libcore/FileUtils.vala]: Spaces before open brackets (3x)

File: libcore/FileUtils.vala [hunk 3]
```diff
@@ -206,27 +206,58 @@ namespace PF.FileUtils {
...
+                        if (paths.length > 1) {
+                            sb.append(Path.DIR_SEPARATOR_S);
+                            sb.append(paths[1]);
```
Comment (jeremypw) [inline libcore/FileUtils.vala]: Spaces before open brackets (2x)

File: libcore/FileUtils.vala [hunk 4]
```diff
@@ -206,27 +206,58 @@ namespace PF.FileUtils {
...
+                        }
+                        break;
+                    // process current dir
+                    case ".":
+                        sb.assign(cp);
+                        if (paths.length > 1) {
+                            sb.append(Path.DIR_SEPARATOR_S);
+                            sb.append(paths[1]);
```
Comment (jeremypw) [inline libcore/FileUtils.vala]: Spaces before open brackets.

File: libcore/FileUtils.vala [hunk 5]
```diff
@@ -206,27 +206,58 @@ namespace PF.FileUtils {
...
+                        }
+                        break;
+                    // process directory without root
+                    default:
+                        sb.assign(cp);
+                        sb.append(Path.DIR_SEPARATOR_S);
+                        sb.append(paths[0]);
+                        if (paths.length > 1) {
+                            sb.append(Path.DIR_SEPARATOR_S);
+                            sb.append(paths[1]);
```
Comment (jeremypw) [inline libcore/FileUtils.vala]: Spaces before open brackets (2x).

File: libcore/FileUtils.vala [hunk 6]
```diff
@@ -292,9 +323,9 @@ namespace PF.FileUtils {
             protocol = "";
         }
 
-        if (!new_path.has_prefix (Path.DIR_SEPARATOR_S)) {
-            new_path = Path.DIR_SEPARATOR_S + new_path;
-        }
+        //  if (!new_path.has_prefix (Path.DIR_SEPARATOR_S)) {
+        //      new_path = Path.DIR_SEPARATOR_S + new_path;
+        //  }
```
Comment (jeremypw) [inline libcore/FileUtils.vala]: Remove commented out code.

**Comment by Janiczek** [comment]
Thank you! <3

**Pull request merged**